linera-service/src/server.rs

```diff
@@ -148,25 +151,32 @@ impl ServerContext {
                 }
             });
         }
+
         join_all(handles).await;
 
         Ok(())
     }
 
-    fn start_metrics(host: &str, port: u16) {
+    fn start_metrics(host: &str, port: &u16) {
         match format!("{}:{}", host, port).parse::<SocketAddr>() {
-            Err(err) => error!("Invalid metrics address: {err}"),
-            Ok(address) => {
-                if let Err(error) = metrics_exporter_tcp::TcpBuilder::new()
-                    .listen_address(address)
-                    .install()
-                {
-                    error!(
-                        ?error, %address,
-                        "Could not install TCP metrics exporter."
-                    );
-                }
-            }
+            Err(err) => error!("Invalid metrics address for {host}:{port}: {err}"),
+            #[cfg(not(feature = "prometheus-metrics"))]
+            Ok(address) => Self::start_metrics_impl(address),
+            #[cfg(feature = "prometheus-metrics")]
+            Ok(address) => prometheus_server::start_metrics(address),
+        }
+    }
+
+    #[cfg(not(feature = "prometheus-metrics"))]
+    fn start_metrics_impl(address: SocketAddr) {
+        if let Err(error) = metrics_exporter_tcp::TcpBuilder::new()
```

Comment:
Why do we want to keep going if metrics don't work?

Reply:
That's probably my bad.
(I think back then we assigned the same metrics port to all servers in integration tests, so that would have made them fail.)